app/tests/spec/lib/metrics.js

```diff
@@ -112,95 +114,233 @@ function (chai, $, p, Metrics, AuthErrors, WindowMock, TestHelpers) {
+            assert.strictEqual(data.broker, 'none');
+            assert.isArray(data.ab);
+          });
+
+          it('resolves to true', function () {
+            assert.isTrue(result);
+          });
+
+          it('clears the event stream', function () {
+            assert.strictEqual(metrics.getFilteredData().events.length, 0);
+          });
+        });
 
-        return metrics.flush()
-            .then(function (data) {
-              var parsedSentData = JSON.parse(sentData);
-              assert.deepEqual(data, parsedSentData);
+        describe('sendBeacon fails', function () {
+          var result;
 
-              var events = metrics.getFilteredData().events;
-              assert.equal(events.length, 0);
+          beforeEach(function (done) {
+            sandbox.stub(windowMock.navigator, 'sendBeacon', function () {
+              return false;
             });
+            metrics.flush().then(function (r) {
+              result = r;
+              done();
+            }, done);
+          });
+
+          afterEach(function () {
+            result = undefined;
+          });
+
+          it('resolves to false', function () {
+            assert.isFalse(result);
+          });
+
+          it('does not clear the event stream', function () {
+            assert.strictEqual(metrics.getFilteredData().events.length, 2);
+          });
+        });
       });
 
-      it('sends filtered data to the server on window unload', function (done) {
-        metrics.logEvent('event10');
-        metrics.logEvent('event20');
+      describe('does not have sendBeacon', function () {
+        beforeEach(function () {
+          sandbox.stub(environment, 'hasSendBeacon', function () {
+            return false;
+          });
+        });
 
-        var filteredData = metrics.getFilteredData();
-        metrics.on('flush.success', function () {
-          var parsedSentData = JSON.parse(sentData);
+        describe('ajax succeeds', function () {
+          var result;
 
-          // `duration` fields are different if the above `getFilteredData`
-          // is called in a different millisecond than the one used to
-          // generate data that is sent to the server.
-          // Ensure `duration` is in the results, but do not compare the two.
-          assert.isTrue(parsedSentData.hasOwnProperty('duration'));
+          beforeEach(function (done) {
+            sandbox.stub(xhr, 'ajax', function () {
+              return p(true);
+            });
+            metrics.logEvent('baz');
+            metrics.flush().then(function (r) {
+              result = r;
+              done();
+            }, done);
+          });
+
+          afterEach(function () {
+            result = undefined;
+          });
+
+          it('calls ajax correctly', function () {
+            assert.isTrue(xhr.ajax.calledOnce);
+            assert.lengthOf(xhr.ajax.getCall(0).args, 1);
+            assert.isObject(xhr.ajax.getCall(0).args[0]);
```

Comment:
Since `args[0]` is referenced so often, the immediate question that pops into my head is "what is args[0]"? Maybe assign it to a descriptive variable to make it slightly easier to read?